doc/src/arch/reference.rst

```diff
@@ -1897,6 +1902,98 @@ The clocking configuration is specified with ``<clock>`` tags within the ``<cloc
+            <!-- connections from spine to rib -->
+            <tap from="spine1.tap" to="rib1.drive" switch="general_routing_switch" fc_val="0.5"/>
+            <!-- connections from rib to clock pins -->
+            <tap from="rib1.tap" to="CLOCK" switch="ipin_cblock" fc_val="1.0"/>
+        </clock_routing >
+    </clocknetworks >
+
+The ``<metal_layers>`` element describes the per unit length electrical parameters used to implement the clock distribution  wires. There can be one or more wiring implementation options (metal layer, width and spacing) that are used by the later  clock network specification and each is described in a separate ``<metal_layer>`` sub-element. The syntax of the wiring electrical information is:
+
+.. arch:tag:: <metal_layer name="string" Rmetal="float" Cmetal="float"/>
+
+    :req_param name: A unique string for reference.
+    :req_param Rmetal: The resistance in Ohms of the wire per unit block in the FPGA architecture; a unit block usually corresponds to a logic cluster.
+    :req_pram Cmetal: The capacitance in Farads of the wire per unit block.
+
+The ``<clock_network>`` element contains sub-elements that describe the clock distribution wires for the clock architecture.  There could be more than one ``<clock_network>`` element to describe separate types of distribution wires. The high-level start tag for a clock network is as follows:
+
+.. arch:tag:: <clock_network name="string" num_inst="integer">
+
+    :req_param name: A unique string for reference.
+    :req_param num_inst: which describes the number of parallel instances of the clock distribution types described in the ``<clock_network>`` sub-elements.
+
+The supported clock distribution types are ``<spine>`` and ``<rib>``. ``Spine`` is used to describe vertical clock distribution wires. Whereas, ``Rib`` is used to describe a horizontal clock distribution wire.
+
+.. arch:tag:: <spine metal_layer="string" x="integer" starty="integer" endy="integer" repeatx="integer" repeaty="integer"/>
+    :req_param metal_layer: A referenced metal layer that sets the unit resistance and capacitance of the distribution wire over the length of the wire.
+    :req_param starty: The start y grid location, of the wire which runs parallel to the y-axis from starty and ends at endy, inclusive. Value can be relative to the device size.
+    :req_param endy: The end of y grid location of the wire. Value can be relative to the device size.
+    :req_param x: The location of the spine with respect to the x-axis. Value can be relative to the device size.
+    :opt_param repeatx: The horizontal repeat factor of the spine along the device. Value can be relative to the device size.
+    :opt_param repeaty: The vertical repeat factor of the spine along the device. Value can be relative to the device size.
+
+The provided example clock network defines two spines, and neither repeats as each spans the entire height of the device and is locally at the horizontal midpoint of the device.
```

Comment:
It would be good to add a cross-reference back to the example you are referring to. Likely means you'll need to label the example code block so you can `:ref:` it.